### PR TITLE
[hw] Add the support for ClickHouse in SQL blocks

### PR DESCRIPTION
**Pull request by csharplus**
# Summary

Integrate with ClickHouse in SQL blocks.

# Tests

This change has been tested with a locally installed ClickHouse on Mac OS. The setup procedure of ClickHouse 
was following [Quick Install](https://clickhouse.com/docs/en/install), i.e., 
```
$ curl https://clickhouse.com/ | sh
$ ./clickhouse server    # in one terminal
$ ./clickhouse client    # in another terminal
```

Verifying the ClickHouse setup with the following SQL command in the Client prompt:
```
:) SELECT 1

SELECT 1

┌─1─┐
│ 1 │
└───┘

1 rows in set. Elapsed: 0.003 sec.

:)
```

The Python library used for connecting with ClickHouse is [clickhouse-connect](https://github.com/ClickHouse/clickhouse-connect) provided by `ClickHouse.com`. More details about the usage of this Python library is available at [Python Integration with ClickHouse Connect](https://clickhouse.com/docs/en/integrations/python) .

Validation through Mage UI was done through the following steps:

1. Initiate and start the local Mage development environment with the following commands:
```
$ ./scripts/init.sh dev_test
$ ./scripts/dev.sh dev_test
```

2. Open the Mage UI at: http://localhost:3000/

3. Create a new Standard (batch) pipeline, and add a SQL `Data loader` block:
Screenshot 2023-04-13 at 10 33 17 PM

4. Add the following settings to `io_config.yaml`:
```
  CLICKHOUSE_DATABASE: default
  CLICKHOUSE_HOST: host.docker.internal
  CLICKHOUSE_INTERFACE: http
  CLICKHOUSE_PASSWORD: null
  CLICKHOUSE_PORT: 8123
  CLICKHOUSE_USERNAME: null
```

5. In the SQL Block editor, choose `Clickhouse` under `Connection` and `Default` under `Profile`, then run the following statement:
```
select 1;
```
Results similar to the following would be returned:
(1) If `Use raw SQL` is checked:
Screenshot 2023-04-13 at 10 26 36 PM

(2) If `Use raw SQL` is unchecked:
Screenshot 2023-04-13 at 10 27 16 PM

6. A more complicated validation with multiple queries could be run as well:
```
CREATE OR REPLACE TABLE test
(
    id UInt64,
    updated_at DateTime DEFAULT now(),
    updated_at_date Date DEFAULT toDate(updated_at)
)
ENGINE = MergeTree
ORDER BY id;

INSERT INTO test (id) Values (1);

SELECT * FROM test;
```
The results similar to the following would be returned:
Screenshot 2023-04-13 at 10 44 11 PM

cc:

@dy46 @wangxiaoyou1993 @tommydangerous

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Could you test this PR in Mage UI?

**Comment by csharplus** [comment]
> Could you test this PR in Mage UI?

Sure, will do.

**Comment by csharplus** [comment]
@wangxiaoyou1993 I've added some details regarding local testing. Please review and see if it is Ok to merge.

File: mage_ai/api/resources/DataProviderResource.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 DATA_PROVIDERS = [
     DataSource.BIGQUERY,
+    DataSource.CLICKHOUSE,
```
Comment (bun4uk) [inline mage_ai/api/resources/DataProviderResource.py:11]: lgtm

**Review by bun4uk — APPROVED** [review]
lgtm

**Review by wangxiaoyou1993 — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -51,6 +51,9 @@ def readme():
             'google-cloud-bigquery~=3.0',
             'db-dtypes==1.0.5',
         ],
+        'clickhouse': [
+            'clickhouse-connect==0.5.20',
```
[the remark this answers is not in the document]
Comment (csharplus) [reply setup.py:55]: > could you add this package to the "all" list as well?

Good catch: made the suggested change.

**Comment by csharplus** [comment]
@wangxiaoyou1993 I don't have Merge button enabled somehow. Could you please help me merge it?

**Comment by wangxiaoyou1993** [comment]
> @wangxiaoyou1993 I don't have Merge button enabled somehow. Could you please help me merge it?

Yes!

**Pull request merged**